Automattic/wp-calypso
# Checkout: Attempt to auto-complete payment method step for Jetpack and Akismet

There are currently two steps in checkout: the billing info step (sometimes even that does not exist for certain free purchases) and the payment method step. Each `CheckoutStep` as defined by the `@automattic/composite-checkout` package has the ability to be rendered in an "active" and an "inactive" state. For example, when you enter your billing info into the first step and press "Continue", the step is changed to its "inactive" state which displays the billing info without the editable fields.

We have a feature which attempts to auto-complete the billing info step using cached contact details when checkout loads, but there is no way to complete the payment method step because a `CheckoutStep` will not display the "Continue" button for the last step. This is so it doesn't conflict with the `CheckoutFormSubmit` which is the page's primary call-to-action.

In https://github.com/Automattic/wp-calypso/pull/83985, we modified the auto-complete feature to also attempt to complete the payment method step when checkout loads. Since this is done programmatically, it allows marking the step as inactive even though there's no manual way to do so. This was A/B tested successfully and was fully launched in https://github.com/Automattic/wp-calypso/pull/85516 for WPCom checkout.

## Proposed Changes

In this diff, we remove the feature gating for the collapse feature altogether, which will have the effect of launching the feature for non-wpcom checkout (Akismet and Jetpack).

This is part of the checkout redesign: https://github.com/Automattic/payments-shilling/issues/1969

## Screenshots

Before             |  After
:-------------------------:|:-------------------------:
<img width="834" alt="before-85797" src="https://github.com/Automattic/wp-calypso/assets/2036909/fba23233-4b66-467c-9b4e-3b3b5b4e5e39"> | <img width="836" alt="after-85797" src="https://github.com/Automattic/wp-calypso/assets/2036909/baa9883b-01cd-4a98-87e8-0fbe33d865d0">


## Testing Instructions

- Using an account which has a saved credit card, add an Akismet or Jetpack product to your cart and visit checkout.
- Verify that the payment method step is collapsed and the card is pre-selected.
